demo/nbody/src/main.rs

```diff
@@ -1,111 +1,123 @@
-        if arg == "--no-par" {
-            run_par = false;
-        } else if arg == "--no-seq" {
-            run_seq = false;
-        } else if arg == "--visualize" {
-            visualize = true;
-        } else if arg == "--ticks" {
-            if let Some(ticks_arg) = args.next() {
-                match usize::from_str(&ticks_arg) {
-                    Ok(v) => ticks = v,
-                    Err(_) => {
-                        println!("invalid argument `{}`, expected integral number of ticks",
-                                 ticks_arg);
-                    }
-                }
-            } else {
-                println!("missing argument to --ticks");
-            }
-        } else if arg == "--bodies" {
-            if let Some(bodies_arg) = args.next() {
-                match usize::from_str(&bodies_arg) {
-                    Ok(v) => num_bodies = v,
-                    Err(_) => {
-                        println!("invalid argument `{}`, expected integral number of bodies", arg);
-                    }
-                }
-            } else {
-                println!("missing argument to --bodies");
-            }
-        } else if arg == "--help" {
-            println!("Usage:");
-            println!(" --no-par      skip parallel execution");
-            println!(" --no-seq      skip sequential execution");
-            println!(" --bodies N    use N bodies (default: {})", DEFAULT_NUM_BODIES);
-            println!(" --ticks N     simulate for N ticks (default: {})", DEFAULT_TICKS);
-            return;
-        } else {
-            println!("Illegal argument `{}`. Try --help.", arg);
-            return;
-        }
+    let args: Args =
+        Docopt::new(USAGE)
+            .and_then(|d| d.decode())
+            .unwrap_or_else(|e| e.exit());
+
+    if args.cmd_benchmark {
+        run_benchmarks(!args.flag_no_par, !args.flag_no_seq,
+                       args.flag_bodies, args.flag_ticks);
     }
 
-    if visualize {
-        visualize_benchmarks(num_bodies);
-    } else {
-        run_benchmarks(run_par, run_seq, num_bodies, ticks);
+    if args.cmd_visualize {
+        visualize_benchmarks(args.flag_bodies);
     }
 }
 
-fn run_benchmarks(run_par: bool,
-                  run_seq: bool,
-                  num_bodies: usize,
-                  ticks: usize) {
-    println!("Configuration:");
-    if !run_par { println!("  --no-par"); }
-    if !run_seq { println!("  --no-seq"); }
-    println!("  --bodies {}", num_bodies);
-    println!("  --ticks {}", ticks);
-
-    let par_time = {
+fn run_benchmarks(run_par: bool, run_seq: bool, bodies: usize, ticks: usize) {
+    let par_time = if run_par {
         let mut rng = XorShiftRng::from_seed([0, 1, 2, 3]);
-        let mut benchmark = NBodyBenchmark::new(num_bodies, &mut rng);
+        let mut benchmark = NBodyBenchmark::new(bodies, &mut rng);
         let par_start = time::precise_time_ns();
+
+        let mut progress = ProgressBar::new(ticks);
```

Comment:
Hmm, you might be right. Although `ticks` is quite small. I can remove it though, just felt nicer to have something to look at :)